classes/UserReputation.php

```diff
@@ -0,0 +1,431 @@
+     *
+     * Decreases user rep by amount provided
+     * If $decrement_weekly is enabled, this will also decrease weekly rep allowing for restricted methods to allow gains again
+     */
+    public function subtractRep(int $amount, bool $decrement_weekly = false) {
+        $this->rep -= $amount;
+        //TODO: TEMPORARY! Remove with negative reputation (outlaw update)
+        if($this->rep < 0) {
+            $this->rep = 0;
+        }
+        if($decrement_weekly && $this->weekly_rep > 0) {
+            $this->weekly_rep -= $amount;
+            if($this->weekly_rep < 0) {
+                $this->weekly_rep = 0;
+            }
+        }
+    }
+
+    // Returns numeric value of reputation
+    public function getRepAmount():int {
+        return $this->rep;
+    }
+    // Returns numeric value of weekly reputaiton
+    public function getWeeklyRepAmount():int {
+        return $this->weekly_rep;
+    }
+    // Return of user can gain more rep for restricted methods
+    public function canGain($check_mission_cd = false):bool {
+        if($this->weekly_rep < $this->weekly_cap) {
+            if($check_mission_cd && $this->mission_cd - time() > 0) {
+                return false;
+            }
+            return true;
+        }
+        return false;
+    }
+    // Load reputation benefits
+    private function loadBenefits() {
+        $benefits = self::$Benefits;
+
+        // Active benefits based on rank
+        switch($this->rank) {
+            case 9:
+            case 8:
+            case 7:
+                $benefits[self::BENEFIT_PARTIAL_TRAINING_GAINS] = true;
+            case 6:
+                $benefits[self::BENEFIT_EFFICIENT_LONG] = true;
+                $benefits[self::BENEFIT_EFFICIENT_EXTENDED] = true;
+            case 5:
+                $benefits[self::BENEFIT_JUTSU_SCROLL_DISCOUNT] = true;
+            case 4:
+                $benefits[self::BENEFIT_GEAR_DISCOUNT] = true;
+            case 3:
+                $benefits[self::BENEFIT_CONSUMABLE_DISCOUNT] = true;
+            case 2:
+            case 1:
+        }
+
+        return $benefits;
+    }
+
+    // Calculate and return reputation amount gain from arena fights
+    public function calcArenaReputation($player_level, $opponent_level) {
+        if($player_level > $opponent_level) {
+            if($player_level - $opponent_level >= 2) {
+                return 0;
+            }
+            return 1;
+        }
+        else {
+            $level_difference = $opponent_level - $player_level;
+            if($level_difference <= 2) {
+                return 1;
+            }
+            else {
+                return 2;
+            }
+        }
+    }
+    // Calculate and return reputation gains/losses from pvp wins/losses
+    public function calcPvpRep($player_level, $player_rep_rank, $opponent_level, $opponent_rep_rank, $opponent_user_id, $winner = true) {
+        $level_difference = $player_level - $opponent_level;
```

Comment:
```suggestion
        $player_levels_above_opponent = $opponent_level - $player_level;
```


This is backwards, below the comments state that $level_difference = -5 means that the opponent is 5 levels below player.

However if player level is 75 and opponent level is 70, the value of $level_difference is 75 - 70, or +5